### PR TITLE
Updates draft on the way to version -04

### PR DESCRIPTION
**Pull request by raphaelvrosa**
Missing VNF-BD/VNF-PP Yang models in Appendix.

### CODE REVIEW

File: draft-rosa-bmwg-vnfbench [hunk 1]
```diff
@@ -144,6 +147,29 @@ Internet-Draft                  VNFBench                   December 2018
...
+   existence of a myriad of automating tools and open interfaces to
+   programmatically manage VNFs; (iii) the existent means to construct
+   syntetic traffic workloads purely by software; (iv) the increasing
+   availability of datasets containing actual sources of production
+   traffic able to be reproduced in benchmarking tests; (v) the varied
+   set of orchestration platforms enabling the allocation of resources
+   and instantition of VNFs through automated machineries based on well-
+   defined templates; (vi) the ability to utilize a large tool set of
+   software components to compose pipelines that mathematically analyze
+   benchmarking metrics in automated ways.
```
Comment (mpeuster) [inline draft-rosa-bmwg-vnfbench:162]: And I am not sure about the purpose of the paragraph? I mean sure those technical factors enable NFV benchmarking. But why do we need this in the intro?

Based on the feedback of the last meeting, I would says we should even try to make the draft "shorter/lighter". Maybe I am missing the point here. 

And aren't (i), (iii), (iv) somehow similar?

File: draft-rosa-bmwg-vnfbench [hunk 2]
```diff
@@ -270,72 +294,104 @@ Internet-Draft                  VNFBench                   December 2018
...
-      machine learning algorithms).
 
-   For the purposes of dissecting the automated execution procedures,
-   consider the following definitions:
+   (e.g., OpenStack, Kubernetes, Open Source MANO).  In automated
+   manners, the realization of a benchmarking testbed/scenario through
+   those means usually rely on network service templates (e.g., TOSCA,
+   Heat, YANG).  Such descriptions must capture relevant details of the
+   execution environment enabling the benchmarking testbed that might
+   influence the whole set of measurements taking place along a test.
```
Comment (mpeuster) [inline draft-rosa-bmwg-vnfbench:346]: This last sentence does not make sense to me.

Maybe something like "Such descriptors have to capture all relevant details of the execution environment to allow the benchmarking framework to correctly instantiate the SUT as well as helper functions, like probes." or so.

File: draft-rosa-bmwg-vnfbench [hunk 3]
```diff
@@ -270,72 +294,104 @@ Internet-Draft                  VNFBench                   December 2018
...
-   Method:   Consists of one or more Tests to benchmark a VNF.  A Method
-      can explicitly list ranges of parameter values for the
-      configuration of benchmarking components.  Each value of such a
-      range is to be realized in a Test.  I.e., Methods can define
-      parameter studies.
+4.2.3.  Execution
 
+   The execution of benchmarking tests can occur repeatedly according to
+   configuration and orchestrated components for each VNF benchmarking
+   Test.  During a test execution, particular behaviors of a VNF can be
```
Comment (mpeuster) [inline draft-rosa-bmwg-vnfbench:365]: Test vs. test

File: draft-rosa-bmwg-vnfbench [hunk 4]
```diff
@@ -270,72 +294,104 @@ Internet-Draft                  VNFBench                   December 2018
...
+   execution environment metrics are available for extraction, Tests
+   should be run to verify if the monitoring of the VNF and/or execution
+   environment can impact the performance measurements.
 
+4.2.4.  Report
 
+   The report of a VNF benchmarking might contain generic VNF footprint
+   metrics (e.g., CPU and memory consumption) and specific VNF traffic
+   processing metrics (e.g., transactions or throughput), which can be
+   parsed and processed in generic or specific ways (e.g., by statistics
```
Comment (mpeuster) [inline draft-rosa-bmwg-vnfbench:381]: ~~parsed~~ stored

File: draft-rosa-bmwg-vnfbench [hunk 5]
```diff
@@ -611,88 +725,74 @@ Internet-Draft                  VNFBench                   December 2018
...
+6.1.5.2.  Links
+
+   Information about the data plane links interconnecting the components
+   of a benchmarking setup.  Links refer to two or more connection-
+   points of a node.  A link might refer to be part of a network.
+   Depending on the link type, the network might be implemented as a
+   layer 2 mesh, or as directional-oriented traffic forwarding flow
+   entries.  Links also detain resource requirements, specifying the
+   minimum bandwidth, latency, and frame loss rate for the execution of
+   benchmarking Tests.
```
Comment (mpeuster) [inline draft-rosa-bmwg-vnfbench:780]: Here People might ask why we specify the topology of the test in the BD itself, and do not simply re-use existing VNF/SFC description approaches as used by OSM/ONAP etc? Do we have an answer for this?

File: draft-rosa-bmwg-vnfbench [hunk 6]
```diff
@@ -1110,19 +1141,7 @@ Internet-Draft                  VNFBench                   December 2018
    also VNF Benchmarking Descriptor (VNF-BD) examples on the vIMS and
    OVS targets as described in [Rosa-b] and [Rosa-c].
```
Comment (mpeuster) [inline draft-rosa-bmwg-vnfbench:1142]: How about the code availability. I think we really need this when we hand in the updated draft.

**Review by mpeuster — COMMENTED** [review]
I did the review.

File: draft-rosa-bmwg-vnfbench (hunk 1 above)
Comment (raphaelvrosa) [reply draft-rosa-bmwg-vnfbench:162]: Change this paragraph to open source section.

File: draft-rosa-bmwg-vnfbench (hunk 2 above)
Comment (raphaelvrosa) [reply draft-rosa-bmwg-vnfbench:346]: Rewrite setence.

File: draft-rosa-bmwg-vnfbench (hunk 3 above)
Comment (raphaelvrosa) [reply draft-rosa-bmwg-vnfbench:365]: Rewrite the whole draft formating as small/capital letters for Test/Trial/Method terms.

File: draft-rosa-bmwg-vnfbench (hunk 4 above)
Comment (raphaelvrosa) [reply draft-rosa-bmwg-vnfbench:381]: Agreed!

File: draft-rosa-bmwg-vnfbench (hunk 5 above)
Comment (mpeuster) [reply draft-rosa-bmwg-vnfbench:780]: Maybe make clear that topologies can either be defined BD or alternativly a topology descriptions can be referenced (external node concepts
)

Comment (raphaelvrosa) [reply draft-rosa-bmwg-vnfbench:780]: Rewrite the sentence to have a clear view of the external node being specified by an OSM/ONAP template.

File: draft-rosa-bmwg-vnfbench (hunk 6 above)
Comment (raphaelvrosa) [reply draft-rosa-bmwg-vnfbench:1142]: Solved!

**Comment by raphaelvrosa** [comment]
Comments being handled by another pull request.

**Pull request closed without merging**